src/mlpack/methods/ann/ffn.hpp

```diff
@@ -142,6 +142,16 @@ class FFN
   void Predict(arma::mat& predictors, arma::mat& results);
 
   /**
+   * Predict the responses to a given set of predictors. The responses will
+   * reflect the output of the given output layer as returned by the
+   * output layer function.
+   *
+   * @param predictors Input predictors.
+   * @return Output predictions of responses into.
+   */
+  arma::mat Predict(arma::mat predictors);
+
```

Comment:
The non-const version would be an rvalue reference (`mat&&`) not a non-const reference (`mat&`).  What this means is that if I want to avoid a copy and I don't need those predictors back, I can do

```
network.Predict(std::move(predictors));
```

and now I save time because the predictors didn't need to be copied.  Otherwise I can just use it as normal

```
network.Predict(predictors);
```

and a copy will be incurred.  This is the strategy I've been employing throughout other parts of the code, like the trees and the constructors for other models like LogisticRegression, etc.: if you pass with `std::move()`, no copy happens but your data matrix may be lost; otherwise, the data will be copied if needed.